### PR TITLE
[onton-completeness-pt-2] Patch 17: Ad-hoc worktree addition and patch removal

### PR DESCRIPTION
**Pull request by subsetpark**
## Summary
- Add `Add_worktree of string` and `Remove_patch` command variants to `Tui_input.command`
- Parse `"w /path"` to register an existing directory as a worktree for the selected patch via `Worktree.add_existing`
- Parse `"-"` to remove the selected patch from orchestration by marking it merged
- Handle both commands in the TUI input fiber with appropriate validation and logging

## Test plan
- [x] Inline tests for `parse_line` cover new formats (`w /path`, `-`, edge cases)
- [x] `dune build` passes with no warnings
- [x] `dune runtest` passes all tests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: bin/main.ml [hunk 1]
```diff
@@ -292,6 +294,69 @@ let input_fiber ~runtime ~selected ~view_mode ~pr_registry =
...
+                  in
+                  match patch_id_opt with
+                  | None ->
+                      log_event runtime
+                        "Cannot add worktree: no selectable patch"
+                  | Some patch_id -> (
+                      let branch =
+                        Runtime.read runtime (fun snap ->
+                            let agent =
+                              Orchestrator.agent snap.Runtime.orchestrator
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply bin/main.ml]: Fixed. The unsafe `Orchestrator.agent` call (which uses `Map.find_exn`) is no longer needed — the symlink approach doesn't require knowing the branch. The `Worktree.add_existing` call now passes a placeholder branch since only the path matters for the symlink.

File: bin/main.ml [hunk 2]
```diff
@@ -292,6 +294,101 @@ let input_fiber ~runtime ~selected ~view_mode ~pr_registry =
...
+                              .Patch_agent.busy)
+                      in
+                      if busy then
+                        log_event runtime ~patch_id
+                          "Warning: patch is currently running — it may create \
+                           a GitHub PR before stopping";
+                      Runtime.update_orchestrator runtime (fun orch ->
+                          Orchestrator.mark_merged orch patch_id);
+                      log_event runtime ~patch_id
+                        "Patch removed from orchestration")
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply bin/main.ml]: Fixed both issues. Merged patch_id, busy, and merged into a single `Runtime.read` to eliminate the TOCTOU gap. Added an early return with "already merged" message when the selected patch is already merged.
